R/utils.R

```diff
@@ -12,37 +12,22 @@ gen_packrat_lockfile <- function(..., .workdir = file.path( getwd(),"work")){
   return (lockfile)
 }
 
-#' Scan a library folder and build a package list
-#' @param .libdir the library directory to scan
-#' @export
-build_package_list_from_library_dir <- function(.libdir){
-  
-  pkgs <- list();
-  for( file in list.files(.libdir)){
-    version <-  packageDescription(file, lib.loc = .libdir)$Version
-    pkgs[file] <- version
-  }
-  
-  return (pkgs)
-}
-
-
 #' Build a string suitable for passing to gen_runtime_description
-#' @param .pkglist the output from build_package_list_from_library_dir
-#' @param .addversion  TRUE to lock package to version, FALSE to get latest
+#' @param .libdir the library directory to scan
 #' @export
-gen_pkg_desc_from_package_list <- function(.pkglist, .addversion=FALSE){
+#' @importFrom utils installed.packages
+gen_pkg_desc_from_libdir <- function(.libdir, .addversion=FALSE){
   out <- ''
-  
-  pkgnames <- names(.pkglist)
+
+  pkglist <- utils::installed.packages(lib.loc = .libdir)
   add <- FALSE
-  for( name in pkgnames){
+  for( row in 1:nrow(pkglist)){

```

Comment:
just in case there is nothing in pkglist, this would blow up. For any loops where you reach for the construct 1:<something> nrow/length/etc instead the trick is just to use either seq_len or seq_along based on if you're trying to iterate over something (seq_along) or generate a sequence up to a max number you know (seq_len)

In this case `for (row in seq_len(nrow(pkglist))`